src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/MultiplesHelper.java

```diff
@@ -51,46 +51,46 @@ private MultiplesHelper() {
     }
 
     public static List<String> getCaseIds(MultipleData multipleData) {
-
-        if (multipleData.getCaseIdCollection() != null
-                && !multipleData.getCaseIdCollection().isEmpty()) {
-
+        if (collectionNotNullOrEmpty(multipleData.getCaseIdCollection())) {
             return multipleData.getCaseIdCollection().stream()
                     .filter(key -> key.getId() != null && !key.getId().equals("null"))
                     .map(caseId -> caseId.getValue().getEthosCaseReference())
                     .distinct()
                     .collect(Collectors.toList());
 
         } else {
-
             return new ArrayList<>();
-
         }
     }
 
-    // MID EVENTS COLLECTIONS HAVE KEY AS NULL BUT WITH VALUES!
-    public static List<String> getCaseIdsForMidEvent(MultipleData multipleData) {
-
-        if (multipleData.getCaseIdCollection() != null
-                && !multipleData.getCaseIdCollection().isEmpty()) {
-
-            return multipleData.getCaseIdCollection().stream()
-                    .filter(caseId -> caseId.getValue().getEthosCaseReference() != null)
-                    .map(caseId -> caseId.getValue().getEthosCaseReference())
-                    .distinct()
-                    .toList();
+    public static List<String> getCaseIdsFromCollection(List<CaseIdTypeItem> caseIdCollection) {
+        if (!collectionNotNullOrEmpty(caseIdCollection)) {
```

Comment:
Gets a little confusing with double negatives. You could call your function '"collectionHasValues" or something similar. Then you since you check !collectionNotNullOrEmpty more than once, you could have another helper function called collectionIsEmpty which just returns  !collectionNotNullOrEmpty

Reply:
Done